dev/bots/test.dart

```diff
@@ -876,138 +699,260 @@ Future<void> _runFlutterTest(String workingDirectory, {
+  // of the device lab directory.
+
+  Future<void> runGradle1() async {
+    await _runDevicelabTest('build_aar_module_test', environment: gradleEnvironment);
+    if (!Platform.isMacOS) {
+      // TODO(ianh): Fails on macOS looking for "dexdump", https://github.com/flutter/flutter/issues/42494
+      await _runDevicelabTest('gradle_jetifier_test', environment: gradleEnvironment);
+    }
+    await _runDevicelabTest('gradle_non_android_plugin_test', environment: gradleEnvironment);
   }
+
+  Future<void> runGradle2() async {
+    await _runDevicelabTest('gradle_plugin_bundle_test', environment: gradleEnvironment);
+    await _runDevicelabTest('gradle_plugin_fat_apk_test', environment: gradleEnvironment);
+    await _runDevicelabTest('gradle_plugin_light_apk_test', environment: gradleEnvironment);
+  }
+
+  Future<void> runGradle3() async {
+    await _runDevicelabTest('gradle_r8_test', environment: gradleEnvironment);
+    await _runDevicelabTest('module_host_with_custom_build_test', environment: gradleEnvironment);
+    await _runDevicelabTest('module_test', environment: gradleEnvironment);
+  }
+
+  Future<void> runMisc() async {
+    await _runDevicelabTest('dartdocs');
+    if (Platform.isLinux) {
+      await _runDevicelabTest('flutter_create_offline_test_linux');
+    } else if (Platform.isWindows) {
+      await _runDevicelabTest('flutter_create_offline_test_windows');
+    } else if (Platform.isMacOS) {
+      await _runDevicelabTest('flutter_create_offline_test_mac');
+      await _runDevicelabTest('plugin_lint_mac');
+      // TODO(jmagman): Re-enable once flakiness is resolved, https://github.com/flutter/flutter/issues/37525
+      // await _runDevicelabTest('module_test_ios');
+    }
+    await _runDevicelabTest('plugin_test', environment: gradleEnvironment);
+  }
+
+  await selectSubshard(<String, ShardRunner>{
+    'gradle1': runGradle1,
+    'gradle2': runGradle2,
+    'gradle3': runGradle3,
+    'misc': runMisc,
+  });
 }
 
-Future<void> _runDevicelabTest(String testName, {Map<String, String> env}) async {
+Future<void> _runDevicelabTest(String testName, {Map<String, String> environment}) async {
   await runCommand(
     dart,
     <String>['bin/run.dart', '-t', testName],
     workingDirectory: path.join(flutterRoot, 'dev', 'devicelab'),
-    environment: env,
+    environment: environment,
   );
 }
 
-String get androidSdkRoot {
-  final String androidSdkRoot = (Platform.environment['ANDROID_HOME']?.isEmpty ?? true)
-      ? Platform.environment['ANDROID_SDK_ROOT']
-      : Platform.environment['ANDROID_HOME'];
-  if (androidSdkRoot == null || androidSdkRoot.isEmpty) {
-    return null;
-  }
-  return androidSdkRoot;
+void deleteFile(String path) {
+  // There's a race condition here but in theory we're not racing anyone
+  // while this script runs, so should be ok.
+  final File file = File(path);
+  if (file.existsSync())
+    file.deleteSync();
```

Comment:
You can say `recursive: true` and not need the `existsSync()` check

Reply:
Interesting. Done.

Reply:
Looks like you were wrong. :-)
https://cirrus-ci.com/task/6550651382792192?command=main#L104

Reply:
It happens to the best of us.  I even wrote a [test that shows what the behavior is](https://github.com/google/file.dart/blob/master/packages/file/test/common_tests.dart#L2735) 😄